src/ng/rootScope.js

```diff
@@ -848,13 +849,14 @@ function $RootScopeProvider() {
 
         clearPhase();
 
-        while (postDigestQueue.length) {
+        while (postDigestQueuePosition < postDigestQueue.length) {
```

Comment:
No - the `asyncQueue` is drained during the `$digest` cycle where `postDigestQueue` is drained outside of it.  As a result, you can't recurse into it since you'd have gotten a digest in progress error.